revscoring/extractors/api/extractor.py

```diff
@@ -71,24 +75,29 @@ def extract(self, rev_ids, dependents, context=None, caches=None,
         """
         context = context or {}
 
-        if caches is not None:
+        if (caches or cache) is not None:
             logger.debug("Extracting {0} dependents with cache {1}"
-                         .format(len(dependents), caches))
+                         .format(len(dependents), caches or cache))
 
         if hasattr(rev_ids, "__iter__"):
             return self._extract_many(rev_ids, dependents, context, caches,
-                                      profile)
+                                      cache, profile)
         else:
             rev_id = rev_ids
-            cache = caches
+            cache = cache or {}
+            cache.update((caches or {}).get(rev_id, {}))
             return self._extract(rev_id, dependents, context, cache,
                                  profile)
 
-    def _extract_many(self, rev_ids, dependents, context, caches, profile):
+    def _extract_many(self, rev_ids, dependents, context, caches, cache,
```

Comment:
That's right.  The external interface takes a "cache", but internally, we need to do a shallow copy and make a bunch of changes so it is changed to "ex_cache" to avoid naming confusion